app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt

```diff
@@ -0,0 +1,59 @@
+package org.thoughtcrime.securesms.util
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.RectF
+import android.text.style.ReplacementSpan
+
+/**
+ * A Span that draws text with a rounded background.
+ *
+ * @param textColor - The color of the text.
+ * @param backgroundColor - The color of the background.
+ * @param cornerRadius - The corner radius of the background in pixels. Defaults to 8dp.
+ * @param paddingHorizontal - The horizontal padding of the text in pixels. Defaults to 3dp.
+ * @param paddingVertical - The vertical padding of the text in pixels. Defaults to 3dp.
+ */
+
+
+class RoundedBackgroundSpan(
+    private val textColor: Int,
+    private val backgroundColor: Int,
+    private val cornerRadius: Float = toPx(8, Resources.getSystem()).toFloat(), // setting some Session defaults
```

Comment:
This is the first time I see the usage of `Resources.getSystem()`, is it safe to use, in terms of UI scale changing, themeing POV? From the doco I see this:

> Return a global shared Resources object that provides access to only system resources (no application resources), is not configured for the current screen (can not use dimension units, does not change based on orientation, etc), and is not affected by Runtime Resource Overlay.

Reply:
Yeah I was a bit surprised by this as well. That line was there from the original PR and I moved it around, but not sure how good this is. I did test it across multiple themes and it seems fine, but I'm also happy to change it.

Reply:
It seems to be used a fair throughout the app in the existing code.

Reply:
Removed it.